### ISSUE TITLE
Fuzzy Skin Banding Bug when using 'Outside Only"

### ISSUE THREAD

**Issue by DefFudgel** [Ultimaker/Cura#15104]
### Application Version

5.0.0

### Platform

Windows 10

### Printer

Ender

### Reproduction steps

1) Build a model, in this case a cavity inside a solid box, connected by a hex shaped hole. 
![Fuzzy WA 1](https://user-images.githubusercontent.com/129547817/229207269-27165311-28e2-43c9-913e-9e16b3e77deb.PNG)

2) Turn on Fuzzy Skin->"Outside only"

3) Slice

### Actual results

Fuzzy skin is not applied before and after the feature, producing what looks like banding. 

![Fuzzy in hole](https://user-images.githubusercontent.com/129547817/231817198-b54fdba4-73a3-4cbf-a554-c0d3552e1862.PNG)

### Expected results

Fuzzy skin applied consistently on z everywhere except in the feature or hole. 

### Checklist of files to include

- [ ] Log file
- [X] Project file

### Additional information & file uploads

Project File:

[Fuzzy Skin Error example.zip](https://github.com/Ultimaker/Cura/files/11125209/Fuzzy.Skin.Error.example.zip)

**Comment by rburema** [comment]
Cura looks at 2D slices (each horizontal) to determine 'inside' and 'outside' for each layer. Above and below the hole, each layer has an inside (the inner rectangle) and an outside (the outside square). The layers with the hole connect, so there is no 'inside'.

**Comment by rburema** [comment]
> I assume fixing it is a matter of changing the logic on how a hole is defined. [...] I tried to work the logic out on paper, because I don't code, but I couldn't come up with anything that didn't fail in corner cases.

It might also be that this isn't possible to solve in a way that's both logically consistent and that everyone agrees with.

That's not all though: The solution _also_ has to be (relatively) easily explainable to people, and important, but often overlooked requirement.

**Comment by MariMakes** [comment]
Hey @DefFudgel,

Working with masks seems like a clever workaround.  To me, it doesn't seem like a bug in Ultimaker Cura, since what you are seeing is expected behavior because Cura looks at horizontal slices. 
 
Would you still like to keep this issue open or do you agree that we can close it?

**Comment by DefFudgel** [comment]
@MariMakes .  My preference would be to keep it open. The community has seen a few people posting about the problem, and they are also seeing it as an error. 

I do have a possible solution. 

**Assumptions:**
                 Our first objective is to have no fuzzy skin in holes
                 Our second objective is to eliminate the banding effect on the internal planes that looks like a bug

So the expectation is fuzzy skin will only be applied to the maximum XY 'perimeter' of each layer which can be found by:

Calculate a 'center' of a layer

Create a table with starting and ending points for all vectors.

Determine the vector end points that are the maximum distance from the center point at any given angle

Apply fuzzy skin only to those vectors that lay the maximum distance from the center.

Calculating the layer perimeter this way avoids a corner case with layers that have higher layers shadowing them like this. The current way follows the wall to the inside of the model as demonstrated by using a brim:

![image](https://user-images.githubusercontent.com/129547817/236310940-97f4517f-db97-4da6-b67f-9ad3d492dc43.png)

Using the method above would allow us to jump the opening and apply fuzzy only to those vectors that are truly on the 'outside' of the entire model


I love this feature, serious kudos to the person who created it.  I'm developing a few 'out of the box' uses. I'm convinced it is a better starting point if you want to smooth your prints out with filler. It does an amazing job hiding layer lines.

**Comment by MariMakes** [comment]
Hey @DefFudgel,

I took another look at it in Cura 5.3.1 and I was able to resolve it with a modifier mesh!
I think it might help in quite a number of cases. 

1. Load the mesh you want fuzzy skin outside only
2. Resize the mesh to 99.9% 
3. Change the smaller mesh to a cutting Mesh
4. Enable Fuzzy Skin + Fuzzy Skin Outside Only in the print settings
5. Disable Fuzzy Skin + Fuzzy Skin Outside Only for the smaller mesh
6. Have the two meshes overlap 
7. Slice and check the preview

![image](https://github.com/Ultimaker/Cura/assets/40423138/6180b083-ee29-451c-86d9-28c218f2d204)

I hope that helps you get unstuck✌️

I can bring your feature request up to the team to see what they can do to improve it. But I expect that if it's not a quick fix that it will not get a high priority to implement. 
Fingers crossed 🤞
